src/feedback_forensics/app/callbacks.py

```diff
@@ -390,8 +402,32 @@ def _get_default_annotator_cols_config(data) -> str:
             datasets[0], data
         )
         avail_datacol_annotator_names = _get_datacol_annotator_names(datasets[0], data)
+
+        # Load the full dataset (needed to extract all models), which may take a few seconds.
+        # Caching ensures this cost is paid only once.
+        # Previously, we used partial loading (still used by _get_principle_annotator_names
+        # and _get_datacol_annotator_names) to postpone this cost until after configuration,
+        # but full loading now makes those optimizations unnecessary.
```

Comment:
Should we remove these optimizations then completely? Seems like now they actually add additional time rather than save time.

Reply:
Yeah I agree, done!